geomesa-core/src/main/scala/org/locationtech/geomesa/core/filter/package.scala

```diff
@@ -48,20 +48,75 @@ package object filter {
    * @return   A List[ List[Filter] ] where the inner List of Filters are to be joined by
    *           Ands and the outer list combined by Ors.
    */
-  private[core] def logicDistribution(x: Filter): List[List[Filter]] = x match {
-    case or: Or  => or.getChildren.toList.flatMap(logicDistribution)
+  private[core] def logicDistributionDNF(x: Filter): List[List[Filter]] = x match {
+    case or: Or  => or.getChildren.toList.flatMap(logicDistributionDNF)
 
     case and: And => and.getChildren.foldRight (List(List.empty[Filter])) {
       (f, dnf) => for {
-        a <- logicDistribution (f)
+        a <- logicDistributionDNF (f)
         b <- dnf
       } yield a ++ b
     }
 
     case not: Not =>
       not.getFilter match {
-        case and: And => logicDistribution(deMorgan(and))
-        case or:  Or => logicDistribution(deMorgan(or))
+        case and: And => logicDistributionDNF(deMorgan(and))
+        case or:  Or => logicDistributionDNF(deMorgan(or))
+        case f: Filter => List(List(not))
+      }
+
+    case f: Filter => List(List(f))
+  }
+
+  /**
+   * This function rewrites a org.opengis.filter.Filter in terms of a top-level AND with children filters which
+   * 1) do not contain further ANDs, (i.e., ANDs bubble up)
+   * 2) only contain at most one AND which is at the top of their 'tree'
```

Comment:
Doesn't this mean, "only contain at most one OR which is at the top of their 'tree'"?